server/schedulers/balance_leader.go

```diff
@@ -128,8 +128,9 @@ func (conf *balanceLeaderSchedulerConfig) validate() bool {
 func (conf *balanceLeaderSchedulerConfig) Clone() *balanceLeaderSchedulerConfig {
 	conf.mu.RLock()
 	defer conf.mu.RUnlock()
+	var ranges []core.KeyRange
```

Comment:
How about define the capacity for the ranges first?  some difference between copy and append : https://stackoverflow.com/questions/71316774/is-it-equivalent-use-copy-or-append-slice-type-in-golang